enhancements/ingress/optional-ingress.md

```diff
@@ -0,0 +1,274 @@
+- CRDs which are created at runtime if `gateway-api` feature gate is set:
+  - [GatewayClass](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gatewayclasses.yaml)
+  - [Gateway](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_gateways.yaml)
+  - [HTTPRoute](https://github.com/openshift/cluster-ingress-operator/blob/master/assets/gateway-api/gateway.networking.k8s.io_httproutes.yaml)
+
+#### Related APIs
+
+The following APIs are left intact if the ingress capability is disabled:
+- [Route API](https://docs.openshift.com/container-platform/4.13/rest_api/network_apis/route-route-openshift-io-v1.html). Implemented in two places: OpenShift API as [CRD](https://github.com/openshift/api/blob/master/route/v1/route.crd.yaml) and in [openshift-api-server](https://github.com/openshift/openshift-apiserver/tree/master/pkg/route).
+- [Ingress Config API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html). Implemented in OpenShift API as [CRD](https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_ingress.crd.yaml).
+
+However the implementation of these APIs (or parts of them) depend on the ingress operator:
+- The ingress operator creates the default ingress controller to manage the backends for the routes.
+- For [the component routes](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html#spec-componentroutes) the ingress operator creates the required role and role bindings to get the custom TLS certificate secret from `openshift-config` namespace.
+
+#### Console operator
+
+To prevent the console operator from going degraded, [the console route check](https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/healthcheck/controller.go#L122) should be skipped if the ingress capability is not enabled in `ClusterVersion`.
+
+#### Authentication operator
+
+No change to the authentication operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+HyperShift doesn't use the authentication operator.
+
+#### Monitoring operator
+
+No change to the monitoring operator is needed until the implementation of the ingress capability for the standalone OpenShift.
+The monitoring operator skips the alert manager's route check if `ingress` cluster operator is not there.
+
+### Risks and Mitigations
+
+#### Partially implemented capability
+
+The ingress capability will become a part of the capability API. Meanwhile its implementation will be incomplete due to the missing support of the standalone OpenShift. However the usage of the capability is an important stepping stone, it unifies the implementation of the optional ingress for all the OpenShift installation types.
```

Comment:
How do we mitigate this? How will we prevent a user from using the capability in a standalone cluster?

Reply:
> How will we prevent a user from using the capability in a standalone cluster?

That's covered by the installer validation discussed in the earlier *Standalone OpenShift* section.